addons/mail/static/src/messaging/component/follower/follower_tests.js

```diff
@@ -0,0 +1,436 @@
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === 'web/image/res.partner/3/image_128') {
+                return;
+            }
+            return this._super(...arguments);
+        },
+        intercepts: {
+            do_action(ev) {
+                assert.step('do_action');
+                assert.strictEqual(
+                    ev.data.action.res_id,
+                    3,
+                    "The redirect action should redirect to the right res id (3)"
+                );
+                assert.strictEqual(
+                    ev.data.action.res_model,
+                    'res.partner',
+                    "The redirect action should redirect to the right res model (res.partner)"
+                );
+                assert.strictEqual(
+                    ev.data.action.type,
+                    "ir.actions.act_window",
+                    "The redirect action should be of type 'ir.actions.act_window'"
+                );
+            },
+        },
+    });
+    const thread = this.env.entities.Thread.create({
+        id: 100,
+        model: 'res.partner',
+    });
+    const follower = await this.env.entities.Follower.create({
+        partner_id: this.env.session.partner_id,
+        email: "bla@bla.bla",
+        id: 2,
+        is_active: true,
+        is_editable: true,
+        name: "François Perusse",
+    });
+    follower.link({ thread });
+    await this.createFollowerComponent(follower);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Follower').length,
+        1,
+        "should have follower component"
+    );
+    assert.strictEqual(
+        document.querySelectorAll('.o_Follower_details').length,
+        1,
+        "should display a details part"
+    );
+
+    document.querySelector('.o_Follower_details').click();
+    await nextAnimationFrame();
+    assert.verifySteps(
+        ['do_action'],
+        "clicking on follower should redirect to partner form view"
+    );
+});
+
+QUnit.test('click on remove follower', async function (assert) {
+    assert.expect(5);
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === 'web/image/res.partner/3/image_128') {
+                return;
+            }
+            if (route.includes('message_unsubscribe')) {
+                assert.step('message_unsubscribe');
+                return;
+            }
+            return this._super(...arguments);
+        }
+    });
+    const thread = this.env.entities.Thread.create({
+        id: 100,
+        model: 'res.partner',
+    });
+    const follower = await this.env.entities.Follower.create({
+        partner_id: this.env.session.partner_id,
+        email: "bla@bla.bla",
+        id: 2,
+        is_active: true,
+        is_editable: true,
+        name: "François Perusse",
+    });
+    follower.link({ thread });
+
+    // Create a parent component to surround the Follower component in order to be able
```

Comment:
I feel like this test should be on ChatterTopbar or whatever is instantiating Follower. No component should destroy itself and creating a parent here doesn't seem needed then.

Reply:
Yeah but the button remove is within the component so for me it makes sense it destroys itself. What do you think we should do then ?

Reply:
Unit testing the component should just test that the correct event is triggered if that is its expected response on the click of the button.

Testing actual removal from DOM should be done on our real parent component that has the feature.

Reply:
agree with seb